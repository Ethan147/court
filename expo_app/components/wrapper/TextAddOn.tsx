import React from "react";
import { View, Text, StyleProp, TextStyle, ViewStyle } from "react-native";

interface TextAddOnProps {
  component: React.ReactNode;
  value?: string;
  display?: boolean;
  passStyles?: {
    textAddOnContainerView?: StyleProp<ViewStyle>;
    textAddOnTextView?: StyleProp<ViewStyle>;
    textAddOnText?: StyleProp<TextStyle>;
  };
}

const TextAddOn: React.FC<TextAddOnProps> = ({
  component,
  value,
  display,
  passStyles,
}) => {
  const styles = {
    textAddOnContainerView: passStyles?.textAddOnContainerView || {},
    textAddOnTextView: passStyles?.textAddOnTextView || {},
    textAddOnText: passStyles?.textAddOnText || {},
  };

  return (
    <View style={styles.textAddOnContainerView}>
      {component}
      {display && (
        <View style={styles.textAddOnTextView}>
          <Text style={styles.textAddOnText}>{value}</Text>
        </View>
      )}
    </View>
  );
};

export default TextAddOn;
