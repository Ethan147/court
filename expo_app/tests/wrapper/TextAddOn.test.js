import { render } from "@testing-library/react-native";
import React from "react";
import { Text } from "react-native";

import TextAddOn from "../../components/wrapper/TextAddOn";

describe("TextAddOn", () => {
  const TestComponent = <Text>Test Component</Text>;

  it("renders the component and text value correctly", () => {
    const { getByText } = render(
      <TextAddOn component={TestComponent} value="Test Value" display={true} />
    );
    expect(getByText("Test Component")).toBeTruthy();
    expect(getByText("Test Value")).toBeTruthy();
  });

  it("does not render the text value when 'display' prop is false", () => {
    const { getByText, queryByText } = render(
      <TextAddOn component={TestComponent} value="Test Value" display={false} />
    );
    expect(getByText("Test Component")).toBeTruthy();
    expect(queryByText("Test Value")).toBeNull();
  });

  it("snapshot", () => {
    const tree = render(<TextAddOn />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
