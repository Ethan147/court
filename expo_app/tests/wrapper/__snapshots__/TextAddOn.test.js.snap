// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextAddOn snapshot 1`] = `
<View
  style={{}}
/>
`;
